### ISSUE TITLE
Grid rows in DataTemplate with no visible content affect layout in Android ListView

### ISSUE THREAD

**Issue by peternary** [unoplatform/uno#1473]
## Current behavior

Grid rows that contain either a collapsed control or no control still appear to reserve space in a ListView datatemplate, shrinking available space for content-containing rows.

## Expected behavior

Empty/unused grid rows should not reserve any space, as in UWP.

## How to reproduce it (as minimally and precisely as possible)

In the attached project, launch in Android - the first listview has content visibly cut off, while the second listview displays its content fully.  The only difference is the first datatemplate's extra Grid rows containing collapsed controls.
[App14.zip](https://github.com/unoplatform/uno/files/3574731/App14.zip)

## Environment

Nuget Package:
Uno.UI
Package Version(s):
1.46.216-dev.2608
Affected platform(s):
- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

**Comment by MakanWG** [comment]
The collapsed controls don't seem to change anything to the issue.
Using the provided sample project, I noticed that the empty RowDefinitions are enough to replicate the issue : The ListViewItems are still kind of shrinked if you remove the Textblocks.

So investigating more about those I found in the [documentation](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.rowdefinition.height) that RowDefinitions have a default Height of 1* and not 0.
This may be the cause of this issue.

A quick workaround would be to set the RowDefinitions Height to Auto so that they will forcefully be at 0 Height when they have a collapsed Content.
